charts/tekton/templates/ingress.yaml

```diff
@@ -0,0 +1,103 @@
+{{- if .Values.dashboard.ingress.enabled -}}
+# tekton dashboard ingress
+{{ if and .Values.dashboard.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
+  {{- if not (hasKey .Values.dashboard.ingress.annotations "kubernetes.io/ingress.class") }}
+  {{- $_ := set .Values.dashboard.ingress.annotations "kubernetes.io/ingress.class" .Values.dashboard.ingress.className}}
+  {{- end }}
+{{- end }}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
```

Comment:
Do we really need to check with `1.14-0` version? I think we don't have that old clusters. As I know we have clusters, which will use this tekton helm templates of `1.19+` versions only

Reply:
So I think we can simplify it and remove `else if` and `else` conditions here, code will look cleaner

Reply:
that ingress manifest is the default you get when you create the boilerplate using `helm create`. Let's stick as it is, because in future we follow the boilerplate helm offers out of the box. So I just added few lines to ingress.